Source/BoundaryConditions/PML.H

```diff
@@ -187,15 +183,15 @@ public:
 
     ~PML () = default;
 
+    const amrex::Geometry* m_geom;
+    const amrex::Geometry* m_cgeom;
```

Comment:
I think we cannot move those calls outside the `if (patch_type == PatchType::fine)` because based on the result of that `if` condition we also choose to get the `fp` or `cp` pointer for E and B respectively. So this would require the refactoring of the similar functions at the level of Parallelization/WarpXComm.cpp, too. I could that here, or maybe in another separate PR, to keep things simple. Anyways I agree with you that passing the `Geometry` members to a PML function does not make much sense, so I removed that. We still remove code duplication in the PML, even if we leave the double check on the patch type for now.

Reply:
With regard to this thread, I remove some more code duplication in the latest commits. Will leave the rest, if needed and if possible, to future PRs.